rsconf/component/devbox.py

```diff
@@ -98,23 +95,32 @@ def _gen_secrets(self, jc):
         s = super().gen_identity_and_host_ssh_keys(jc, "host", encrypt_identity=True)
         self.secrets = PKDict({k: s[k] for k in ("host_key_f", "identity_pub_f")})
 
-    def _jupyter(self, z, jupyter_bashrc_path):
+    def _jupyter_bashrc(self, z, path):
+        def _env(name, value):
+            v = str(value)
+            # TODO probably could recursively quote
+            if "'" in v:
+                raise ValueError(
+                    f"single quote in value={v} of bash variable name={name}"
+                )
+            self.rsconf_append(
+                path,
+                f"export {name}='{v}'",
+            )
+
+        self.install_access(mode="600")
+        self.install_ensure_file_exists(path)
```

Comment:
I believe this causes the file to be appended to instead of overwritten. Is that desired?

Reply:
Correct. We want to allow local modifications.